File: compiler/bench/pick0.fs
\ expected: 500000000
\ Pick from depth 0 (dup), 500M times
: main
  0                         ( acc )
  500000000 0 do
    0 pick                  ( acc acc ) same as dup
    1 + nip                 ( acc' )
  loop
  . cr
;

File: compiler/bench/pick0.c
// expected: 500000000
// Pick from depth 0 (dup), 500M times
#include <stdio.h>

int main(void) {
    long acc = 0;
    for (long i = 0; i < 500000000L; i++) {
        long picked = acc;  // pick 0 = dup
        acc = picked + 1;
    }
    printf("%ld\n", acc);
    return 0;
}
